ledger-core-ripple/inc/ripple/RippleLikeOperation.hpp

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ * RippleLikeOperation
+ *
+ * Created by El Khalil Bellakrid on 06/01/2019.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <core/operation/Operation.hpp>
+#include <core/operation/OperationQuery.hpp>
+
+#include <ripple/api/RippleLikeOperation.hpp>
+#include <ripple/api/RippleLikeTransaction.hpp>
+
+namespace ledger {
+    namespace core {
+        class RippleLikeOperation : public api::RippleLikeOperation, public Operation {
+        public:
+            RippleLikeOperation() = default;
+            RippleLikeOperation(
+                std::shared_ptr<RippleLikeBlockchainExplorerTransaction> const & tx,
+                api::Currency const & currency
+            );
+            std::shared_ptr<api::RippleLikeTransaction> getTransaction() override;
+
+            void refreshUid() override;
+
+        private:
+            std::shared_ptr<api::RippleLikeTransaction> _transaction;
+        };
+
+        class RippleLikeOperationQuery : public OperationQuery<RippleLikeOperation> {
```

Comment:
I suggest to hold only one class by file - avoid potential confusion about which header should be included.